utils/githooks/commit-msg.d/10-watermark.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+"""Add watermark to commit message"""
+import os
+import sys
+
+def check_if_run(name):
+    """Check existence of a file, removes file"""
+    if not os.path.exists(f".{name}"):
+        print(f"Required githook {name} is not installed. See utils/githooks/README.md")
+        return False
+    os.remove(f".{name}")
+    return True
+
+def find_hooks():
+    """Find the required githooks"""
+    hooks = []
+    for fname in os.listdir('utils/githooks'):
+        if fname == "commit-msg.d":
+            continue
+        if not fname.endswith('.d'):
+            continue
+        if not os.path.isdir(os.path.join("utils/githooks")):
+            continue
+        hooks.append(fname[:-2])
+    return hooks
+
+def run_check():
+    """Run the checks for the required commit hooks"""
+    empty_commit = True
+    msg = []
+    with open(sys.argv[1], "r") as commit_msg:
+        for line in commit_msg.readlines():
+            if line.startswith("Required-githooks: true"):
+                sys.exit(0)
+            elif line.startswith("Signed-off-by"):
+                continue
+            elif line.strip() == "":
+                continue
+            elif line.startswith("#"):
```

Comment:
That is the assumption it makes, yes.    I would presume if someone means to commit something, they would put something in the commit message before the Signed-off-by ?

Reply:
Maybe it's safer and achieves the same goal to `continue` here instead of `break`?

Reply:
The problem here is that if you do git commit -v, the diff gets put after the comments and wouldn't be commented and has the same semantics that it should be ignored because it doesn't get committed that way.

I think it's generally fairly safe to assume that users are only going to add comments at the top before the comments.   I think it's a step forward and we can revisit if we hit issues with it?

Reply:
I'm personally okay with this, but will defer to @brianjmurrell's use case